counsel.el

```diff
@@ -3035,7 +3035,8 @@ Works for `counsel-git-grep', `counsel-ag', etc."
 (defun counsel--grep-smart-case-flag ()
   (if (ivy--case-fold-p ivy-text)
       "-i"
-    (if (string-match-p "\\`pt" counsel-ag-base-command)
+    (if (and (stringp counsel-ag-base-command)
+             (string-match-p "\\`pt" counsel-ag-base-command))
```

Comment:
If `counsel-ag-base-command` is a list, shouldn't the `string-match-p` be against the list's `car`?

Reply:
This branch (`-S` flag) only applies to `pt`, for which `counsel-ag-base-command` is set to a string:
https://github.com/abo-abo/swiper/blob/7fc1a2d5e90d97568a537d02026a68894b391220/counsel.el#L3080-L3083

Reply:
That doesn't matter, and is subject to change.

If `counsel-ag-base-command` can be a list, then it's better and more future-proof if the code handles that case properly.